soc/hps_lattice_nx.py

```diff
@@ -0,0 +1,174 @@
+##################################################################
+# This is a modified version of litex/soc/cores/ram/lattice_nx.py
+# Modified by CFU Playground Authors to:
+#  * optionally use dual port LRAM primitive
+#  * optionally stripe consecutive addresses across the banks
+##################################################################
+#
+#
+# This file is part of LiteX.
+#
+# Copyright (c) 2020 David Corrigan <davidcorrigan714@gmail.com>
+# Copyright (c) 2019 William D. Jones <thor0505@comcast.net>
+# Copyright (c) 2019 Tim 'mithro' Ansell <me@mith.ro>
+# Copyright (c) 2019 Florent Kermarrec <florent@enjoy-digital.fr>
+# SPDX-License-Identifier: BSD-2-Clause
+
+from migen import *
+from litex.soc.interconnect import wishbone
+
+kB = 1024
+
+"""
```

Comment:
For a string to be recognized as a  docstring, it must be the first statement of the module, class or function. See: https://www.python.org/dev/peps/pep-0257/#what-is-a-docstring

Two options:

1. Put the docstring right after the copyright comment, and have it include the expanded explanation you added, ncluding the table.
2. Use an explicit docstring with __doc__ = """\<stuff\>"""

I'd prefer (1)

Reply:
Thanks for pointing that out, updated!